elyra/pipeline/component.py

```diff
@@ -275,73 +236,158 @@ class ComponentReader(LoggingConfigurable):
 
-        with open(component_path, 'r') as f:
+        with open(location, 'r') as f:
             return f.read()
 
+    def get_list_of_paths(self, paths: List[str], parser_file_types: List[str]) -> List[str]:
+        absolute_paths = []
+        for path in paths:
+            absolute_path = self.determine_location(path)
+            if not os.path.exists(absolute_path):
+                self.log.warning(f"File does not exist -> {absolute_path}")
+            absolute_paths.append(absolute_path)
+        return absolute_paths
+
+
+class DirectoryComponentReader(FilesystemComponentReader):
+    """
+    Read component definitions from a local directory
+    """
+    type = 'directory'
+
+    def get_list_of_paths(self, paths: List[str], parser_file_types: List[str]) -> List[str]:
+        absolute_paths = []
+        for path in paths:
+            absolute_path = self.determine_location(path)
+            if not os.path.exists(absolute_path):
+                self.log.warning(f"Invalid directory -> {absolute_path}")
+                continue
+
+            for filename in os.listdir(absolute_path):
+                if filename.endswith(tuple(parser_file_types)):
+                    absolute_paths.append(os.path.join(absolute_path, filename))
+
+        return absolute_paths
+
+    @property
+    def base_type(self):
+        return super().type
+
 
 class UrlComponentReader(ComponentReader):
     """
-    Read a component definition from a url
+    Read a singular component definition from a url
     """
     type = 'url'
 
-    def read_component_definition(self, registry_entry: SimpleNamespace) -> Optional[str]:
+    def read_component_definition(self, location: str) -> Optional[str]:
         try:
-            res = requests.get(registry_entry.location)
+            res = requests.get(location)
         except Exception as e:
-            self.log.warning(f"Failed to connect to URL for component: {registry_entry.id} -> " +
-                             f"{registry_entry.location}: {str(e)}")
+            self.log.warning(f"Failed to connect to URL for component: {location}: {str(e)}")
             return None
 
         if res.status_code != HTTPStatus.OK:
-            self.log.warning(f"Invalid location for component: {registry_entry.id} -> {registry_entry.location} " +
-                             f"(HTTP code {res.status_code})")
+            self.log.warning(f"Invalid location for component: {location} (HTTP code {res.status_code})")
             return None
 
         return res.text
 
+    def get_list_of_paths(self, paths: List[str], parser_file_types: List[str]) -> List[str]:
+        return paths
+
+
+class GitHubComponentReader(UrlComponentReader):
+    """
+    Read component definitions from a github repo
+    """
+    type = 'github'
+
+    def get_list_of_paths(self, paths: List[str], parser_file_types: List[str]) -> List[str]:
+        pass
+
+    @property
+    def base_type(self):
+        return super().type
```

Comment:
Same as Directory, and returns 'url'.  Please add a comment.

Reply:
Since we're adding GitHub location support, we should add `GitHub` to the enum for `location_type` in the schema.